### PR TITLE
Add conversions that UUID types from libraries can be handled by pytype.UUID

### PR DESCRIPTION
**Pull request by NWilson**
This is similar in spirit to all the existing pgtype conversion code.

It's just that arrays (not slices) of primitive types aren't currently handled.

In my case, I was using the popular Google UUID library (https://github.com/google/uuid), which wraps `[16]byte`.

### CODE REVIEW

File: pgtype/convert.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ func underlyingTimeType(val interface{}) (interface{}, bool) {
 	switch refVal.Kind() {
 	case reflect.Ptr:
 		if refVal.IsNil() {
-			return time.Time{}, false
+			return nil, false
```
Comment (jackc) [inline pgtype/convert.go]: This change seems unrelated. Is it intentional?

Comment (NWilson) [reply pgtype/convert.go]: Hi, sorry for the late reply.

Yes, it is intentional, it's in a separate commit. (I could have made a separate PR but it's so minor...) The first return value is ignored when the second is false - so there's no point heap-allocating a zero time.Time that will be thrown away by the caller. It's possibly just a mistake/holdover from when the function's return signature was `(time.Time,bool)`?

**Comment by jackc** [comment]
Thanks!
